### PR TITLE
virtcontainers: add support for loading kernel modules

### LINKED ISSUE #1925: add support for loading kernel modules
use gRPC to pass the list of kernel modules and theirs parametes to the agent, see https://github.com/kata-containers/agent/pull/616

### PR DESCRIPTION
**Pull request by devimc**
The list of kernel modules can be passed to the runtime through the
configuration file or using OCI annotations. In both cases, a list paramentes
can be specified for each module.

fixes #1925

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Comment by devimc** [comment]
waiting for https://github.com/kata-containers/agent/pull/616

File: cli/config/configuration-fc.toml.in [hunk 1]
```diff
@@ -250,6 +250,14 @@ path = "@SHIMPATH@"
 #trace_mode = "dynamic"
 #trace_type = "isolated"
 
+# Comma separated list of kernel modules and their parameters.
+# These modules will be loaded in the guest kernel using modprobe(8).
+# The following example can be used to load two kernel modules with parameters
+#  - kernel_modules=["e1000e InterruptThrottleRate=3000,3000,3000 EEE=1", "i915 enable_ppgtt=0"]
+# The first word is considered as the module name and the rest as its parameters.
+#
+kernel_modules=[]
```
Comment (jodh-intel) [inline cli/config/configuration-fc.toml.in]: This obviously relies on the admin specifying a set of modules that is going to be available in the guests kernel. As such, I think it is worth stating here what happens if:

- the module is not available in the guest kernel.
- `modprobe` disabled in the guest environment.
- the module is architecture-specific and the guest is running an incompatible architecture.

**Comment by devimc** [comment]
@jodh-intel change applied, thanks

**Comment by GabyCT** [comment]
the Power8 failure is not related with this PR

**Pull request merged**